applications/NeuralNetworkApplication/python_scripts/data_generator_process.py

```diff
@@ -0,0 +1,361 @@
+                except IndexError:
+                    pass
+
+            # Writing input file
+            if (self.write_output_file):
+                if self.output_format == "ascii":
+                    self.training_input_file.write(' '.join(str(v) for v in input_value_list) + '\n')
+                if self.output_format == "h5":
+                    self.hdf5_input_parameters["prefix"].SetString("/"+str(current_step)+"/InputData") 
+                    nodal_io = KratosHDF5.HDF5NodalSolutionStepDataIO(self.hdf5_input_parameters, self.hdf5_file_training_input)
+                    nodal_io.WriteNodalResults(self.input_model_parts[0],1)
+
+    def ExecuteFinalizeSolutionStep(self):
+        current_time = self.output_model_parts[0].ProcessInfo[KM.TIME]
+        current_step = self.output_model_parts[0].ProcessInfo[KM.STEP]
+        if((current_time >= self.interval[0]) and (current_time < self.interval[1])):
+            output_value_list=[]
+            for model_part, variable, source in self.dict_output:
+                model_output_value_list = []
+                # Process related variables (e.g. TIME, STEP)
+                if source == 'process':
+                    model_output_value_list.append(model_part.ProcessInfo[variable])
+                # Node properties (e.g. position)
+                elif source == 'node':
+                    for node in model_part.Nodes:
+                        model_output_value_list.append(getattr(node,variable.Name()))
+                # Node step values (e.g. variables like displacement)
+                elif source == "solution_step":
+                    for node in model_part.Nodes:
+                        model_output_value_list.append(node.GetSolutionStepValue(variable,0))
+                # Condition values
+                elif source == "condition":
+                    for condition in model_part.GetConditions():
+                        model_output_value_list.append(condition.GetValue(variable))
+                output_value_list.extend(model_output_value_list)
+            # Reorder if indicated
+            if self.output_order == 'sources_first':
+                try:
+                    output_value_list = self._OrderSourcesFirst(output_value_list, self.dict_output)
+                except IndexError:
+                    pass
+
+            # Writing output file
+            if (self.write_output_file):
+                if self.output_format == "ascii":
+                    self.training_output_file.write(' '.join(str(v) for v in output_value_list) + '\n')
+                if self.output_format == "h5":
+                    self.hdf5_output_parameters["prefix"].SetString("/"+str(current_step)+"/OutputData") 
+                    nodal_io = KratosHDF5.HDF5NodalSolutionStepDataIO(self.hdf5_output_parameters, self.hdf5_file_training_output)
+                    nodal_io.WriteNodalResults(self.output_model_parts[0],1)

```

Comment:
If I am not mistaken, you are restricting the HDF5 output to only `SolutionStep` variables whereas the `ascii` output supports all types. IF so, this can be easily rectified with the `Kratos.Expression` and the revamped HDF5 application. I will put the example usage later. Could you please wait for them?